backend/src/main/java/com/tourneynizer/tourneynizer/model/MatchGenerator.java

```diff
@@ -94,9 +110,22 @@ public int compareTo(MatchNode o) {
             if (node2.getMatch() != null) { match2 = node2.getValueId(); }
             else {                          team2 = node2.getValueId(); }
 
-            MatchChildren children = new MatchChildren(team1, team2, match1, match2);
             int newHeight = Math.max(node1.height, node2.height) + 1;
+
+            MatchChildren children = new MatchChildren(team1, team2, match1, match2);
             Match parent = new Match(tournament.getId(), children, order++, null, ScoreType.ONE_SET, (short) newHeight);
+
+            if (children.getKnownTeamChildren().size() == 2) { // assign ref
+                Long refId = randomizedIds.get(refIndex % randomizedIds.size());
+
+                for (int i = 0; i < 2 && children.getKnownTeamChildren().contains(refId); i++) {
```

Comment:
why i < 2?

Reply:
Each match has two teams, so at most we want to move two teams away to find a team that isn't part of this match. This is to prevent infinite loops, which theoretically should never happen if teamSize > 2, but I'm playing it safe.

Reply:
Ok I understand